test/state/requests.hpp

```diff
@@ -0,0 +1,29 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2024 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "hash_utils.hpp"
+#include <evmc/evmc.hpp>
+#include <span>
+#include <vector>
+
+namespace evmone::state
+{
+/// `requests` object.
+struct Requests
+{
+    /// Request type.
+    uint8_t type = 0;
```

Comment:
is having a default good here?

Reply:
also, maybe introduce a Request::Type enum like for tx?

Reply:
I was meaning to just introduce constants for types in respective later PRs, but enum is a better idea, I'll add it in the next PR.

Having a default in general is good, otherwise it may be left uninitialized.

Reply:
> Having a default in general is good, otherwise it may be left uninitialized.

ah right, that would not error out :(. OK then

Reply:
added enum here